prov/tcp/src/tcpx_ep.c

```diff
@@ -648,7 +648,30 @@ static int tcpx_pep_listen(struct fid_pep *pep)
 	fd_signal_set(&tcpx_fabric->poll_mgr.signal);
 
 	return 0;
+}
+
+static int tcpx_pep_reject(struct fid_pep *pep, fid_t handle,
+		    const void *param, size_t paramlen)
+{
+	struct ofi_ctrl_hdr hdr;
+	struct tcpx_conn_handle *tcpx_handle;
+
+	tcpx_handle = container_of(handle, struct tcpx_conn_handle, handle);
+
+	memset(&hdr, 0, sizeof(hdr));
+	hdr.version = OFI_CTRL_VERSION;
+	hdr.type = ofi_ctrl_nack;
+	hdr.seg_size = paramlen;
 
+	ofi_send_socket(tcpx_handle->conn_fd, &hdr,
+			sizeof(hdr), MSG_NOSIGNAL);
+
+	if (paramlen)
+		ofi_send_socket(tcpx_handle->conn_fd, param,
+				paramlen, MSG_NOSIGNAL);
```

Comment:
I don't think NOSIGNAL is needed here or above.  I'm not sure if this is portable.